gst/nnstreamer/tensor_common.c

```diff
@@ -1168,8 +1168,27 @@ gst_gen_tensors_from_collectpad (GstCollectPads * collect,
   gint old_denominator = G_MAXINT;
   gint counting = 0;
   GstTensorConfig config;
+  GstClockTime base = 0;
 
   walk = collect->data;
+
+  if (sync.mode == SYNC_BASEPAD) {
+    walk = g_slist_nth (walk, sync.data_basepad.sink_id);
+    GstCollectData *data = (GstCollectData *) walk->data;
+    GstTensorCollectPadData *pad = (GstTensorCollectPadData *) data;
+    GstBuffer *buf = NULL;
+    buf = gst_collect_pads_peek (collect, data);
+    if (pad->buffer != NULL && buf != NULL) {
+      base =
+          MIN (sync.data_basepad.duration,
+          ABS (GST_CLOCK_DIFF (GST_BUFFER_PTS (buf),
+                  GST_BUFFER_PTS (pad->buffer))) - 1);
+      gst_buffer_unref (buf);
```

Comment:
possible mem leak if pad->bufer is null and buf is not null. 
please move this line outside, or change the condition to get 'base'.

Reply:
well.. is it matter?? unref if the buf is not null??

Reply:
Good point. Thanks.